src/aggregate/functions/date.c

```diff
@@ -60,19 +60,20 @@ static int timeFormat(ExprEval *ctx, RSValue *result, RSValue **argv, size_t arg
 
 /* Fast alternative to mktime which is dog slow. From:
 https://gmbabar.wordpress.com/2010/12/01/mktime-slow-use-custom-function/ */
+// Fix and performance improvements:
+// https://godbolt.org/z/qscb5d9dT
+// https://quick-bench.com/q/oTV4_9uVqPTcrj2fpDEvbbMzZ48
+// https://quick-bench.com/q/2Bc8WY1Ys0vmbp-HPagWFxu81jI
 time_t fast_timegm(const struct tm *ltm) {
-  // elapsed days until the beginning of every month
-  const int mon_days[] = {0, 31, 59, 90, 120, 151, 181, 212, 243, 273, 304, 334};
-  long tyears, tdays, leaps, utc_hrs;
+  long tyears, tdays, leaps;
```

Comment:
Declaring all variables at the start of a block is an ancient practice that has not been necessary since ANSI C. It is much less error-prone to declare variables at the point of their first use, so that their lifetimes match their use, and you have less risk of dangling.
Additionally, don't declare multiple variables in one statement. That leads to bugs like `int* p, q` (`q` is `int`, not `int*`).